src/main/java/life/qbic/io/commandline/FindDatasetsCommand.java

```diff
@@ -0,0 +1,70 @@
+package life.qbic.io.commandline;
+
+import ch.ethz.sis.openbis.generic.OpenBIS;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.DataSet;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.Person;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import life.qbic.App;
+import life.qbic.model.download.OpenbisConnector;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Mixin;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+@Command(name = "list-data",
+    description = "lists datasets and their details for a given experiment code")
+public class FindDatasetsCommand implements Runnable {
+
+  @Parameters(arity = "1", paramLabel = "experiment", description = "The code of the experiment data is attached to")
+  private String experimentCode;
+  @Option(arity = "1", paramLabel = "<space>", description = "Optional openBIS spaces to filter samples", names = {"-s", "--space"})
+  private String space;
+  @Mixin
+  AuthenticationOptions auth = new AuthenticationOptions();
+
+    @Override
+    public void run() {
+      List<String> spaces = new ArrayList<>();
+      if (space != null) {
+        System.out.println("Querying experiment in space: " + space + "...");
+        spaces.add(space);
+      } else {
+        System.out.println("Querying experiment in all available spaces...");
```

Comment:
NIT: As previously mentioned, maybe a print class with distinct methods makes sense to avoid cluttering the code?